lib/iris/coords.py

```diff
@@ -1628,6 +1628,195 @@ def __hash__(self):
         return hash(id(self))
 
 
+class CellMeasure(six.with_metaclass(ABCMeta, CFVariableMixin)):
+    """
+    A CF Cell Measure, providing area or volume properties of a cell
+    where these cannot be inferred from the Coordinates and
+    Coordinate Reference System.
+
+    """
+
+    def __init__(self, data, standard_name=None, long_name=None,
+                 var_name=None, units='1', attributes=None, measure=None):
+
+        """
+        Constructs a single cell measure.
+
+        Args:
+
+        * data:
+            The values of the measure for each cell.
+
+        Kwargs:
+
+        * standard_name:
+            CF standard name of coordinate
+        * long_name:
+            Descriptive name of coordinate
+        * var_name:
+            CF variable name of coordinate
+        * units
+            The :class:`~cf_units.Unit` of the coordinate's values.
+            Can be a string, which will be converted to a Unit object.
+        * attributes
+            A dictionary containing other cf and user-defined attributes.
+        * measure
+            A string describing the type of measure.  'area' and 'volume'
+            are the only valid entries.
+
+        """
+        #: CF standard name of the quantity that the coordinate represents.
+        self.standard_name = standard_name
+
+        #: Descriptive name of the coordinate.
+        self.long_name = long_name
+
+        #: The CF variable name for the coordinate.
+        self.var_name = var_name
+
+        #: Unit of the quantity that the coordinate represents.
+        self.units = units
+
+        #: Other attributes, including user specified attributes that
+        #: have no meaning to Iris.
+        self.attributes = attributes
+
+        self.data = data
+
+        self.measure = measure
+
+    @property
+    def measure(self):
+        return self._measure
+
+    @property
+    def data(self):
+        """Property containing the data values as a numpy array"""
+        data = self._data
+        if isinstance(data, biggus.Array):
+            data = data.ndarray()
+            self._data = data
+        return data.view()
+
+    @data.setter
+    def data(self, data):
+        # Set the data to a new array - as long as it's the same shape.
+        # If data are already defined for this coordinate,
+        if hasattr(self, '_data') and self._data is not None:
```

Comment:
i've raised a ValueError for the data setter with a payload of None.  _data needs to be able to be set to None, at least Transiently